src/js/common/schemaform/actions.js

```diff
@@ -81,3 +114,182 @@ export function submitForm(formConfig, form) {
+      }
+
+      let status = 'failure';
+      if (res.status === 401) {
+        // TODO: If they've sat on the page long enough for their token to expire
+        //  and try to load, tell them their session expired and they need to log
+        //  back in and try again.
+        status = 'no-auth';
+      } else if (res.status === 404) {
+        status = 'not-found';
+      }
+      return Promise.reject(status);
+    }).then((resBody) => {  // eslint-disable-line consistent-return
+      // Just in case something funny happens where the json returned isn't an object as expected
+      if (typeof resBody !== 'object') {
+        return Promise.reject('invalid-data');
+      }
+
+      // If an empty object is returned, throw a not-found
+      // TODO: When / if we return a 404 for applications that don't exist, remove this
+      if (Object.keys(resBody).length === 0) {
+        return Promise.reject('not-found');
+      }
+
+      // If we've made it this far, we've got valid form
+
+      // TODO: Remove the assignment once the API is updated to store and return
+      //  formData and metadata
+      let formData;
+      try {
+        // NOTE: This may change to be migrated in the back end before sent over
+        formData = getUpdatedFormData(resBody.form_data, resBody.metadata.version, migrations);
+        // formData = getUpdatedFormData(resBody.formData, resBody.metadata.version, migrations);
+      } catch (e) {
+        // TODO: Log e.message somewhere; it's the reason the data couldn't be
+        //  transformed. Sentry error?
+        return Promise.reject('invalid-data');
+      }
+      // Set the data in the redux store
+      // NOTE: Until we get the api for the list of filled forms, we're using this
+      //  function to see if the form has been filled in, so this is setting the
+      //  data in a separate place to be pulled in when we _actually_ want to load
+      //  the formData.
+      // dispatch(setData(formData));
+      dispatch(setLoadedData({ formData, metadata: resBody.metadata }));
+      dispatch(setLoaded('success'));
+    }).catch((status) => {
+      let loadedStatus = status;
+      // if res.json() has a parsing error, it'll reject with a SyntaxError
+      if (status instanceof SyntaxError) {
+        // TODO: Log this somehow...Sentry error?
+        console.error('Could not parse response.', status); // eslint-disable-line no-console
+        loadedStatus = 'invalid-data';
+      }
+      dispatch(setLoaded(loadedStatus));
+    });
+  };
+}
+
+/**
+ * Saves the form data to the back end
+ * @param  {String}  formId    The form's formId
+ * @param  {Ingeter} version   The form's version
+ * @param  {String}  returnUrl The last URL the user was at before saving
+ * @param  {Object}  formData  The data the user has entered so far
+ */
+export function saveFormData(formId, version, returnUrl, formData) {
+  // Double stringify because of api reasons. Olive Branch issues, methinks.
```

Comment:
There was [some chatter](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3046#issuecomment-305308118) about this the other day, but until I'm "officially" told not to, this is working.